### PR TITLE
[sailfishos][gecko] Enable support for WebRTC. JB#55222 OMP#JOLLA-295

### PR DESCRIPTION
**Pull request by adenexter**

### CODE REVIEW

File: rpm/xulrunner-qt5.spec [hunk 1]
```diff
@@ -128,6 +130,11 @@ Patch72:    0072-sailfishos-gecko-Do-not-flip-scissor-rects-when-rend.patch
 Patch73:    0073-sailfishos-components-Cleanup-static-components-defi.patch
 Patch74:    0074-Revert-Bug-1611658-Unship-libcubeb-s-C-PulseAudio-ba.patch
 Patch75:    0075-sailfishos-gfx-Use-scroll-frame-background-color-as-.patch
+Patch76:    0076-sailfishos-webrtc-Adapt-build-configuration-for-Sail.patch
```
[the remark this answers is not in the document]
Comment (adenexter) [reply rpm/xulrunner-qt5.spec:133]: Removed? The configuration changes are mostly input to the makefile generation in the following patch so you can probably build without it, but I don't think removing it's an appropriate response to a conflict from a new patch being added.

File: rpm/0079-sailfishos-webrtc-Enable-GMP-for-encoding-decoding.-.patch [hunk 2]
```diff
@@ -0,0 +1,868 @@
...
++  MutexAutoLock lock(mMutex);
+   for (uint32_t i = 0; i < GetGmpFreelist(aClass).Length(); i++) {
+     MOZ_ASSERT(GetGmpFreelist(aClass)[i].IsWritable());
+     total += GetGmpFreelist(aClass)[i].Size<uint8_t>();
+@@ -86,7 +97,8 @@ bool GMPSharedMemManager::MgrDeallocShmem(GMPSharedMem::GMPMemoryClasses aClass,
+   return true;
+ }
+ 
+-uint32_t GMPSharedMemManager::NumInUse(GMPSharedMem::GMPMemoryClasses aClass) {
++int32_t GMPSharedMemManager::NumInUse(GMPSharedMem::GMPMemoryClasses aClass) {
```
[the remark this answers is not in the document]
Comment (adenexter) [reply rpm/0079-sailfishos-webrtc-Enable-GMP-for-encoding-decoding.-.patch:81]: It was like that in the original patch and I don't know why. The function implementation should never return a negative number so I too considered changing it but fell into the what am I not seeing trap.

**Review by d-grigorev — APPROVED** [review]

File: rpm/0079-sailfishos-webrtc-Enable-GMP-for-encoding-decoding.-.patch (hunk 2 above)
[the remark this answers is not in the document]
Comment (d-grigorev) [reply rpm/0079-sailfishos-webrtc-Enable-GMP-for-encoding-decoding.-.patch]: I remember I had an issue with these counters, but perhaps it was before I added a mutex there.

**Review by rainemak — APPROVED** [review]
Thanks @d-grigorev for your review. Approving as well. Please rebase before merging if there's need.

**Pull request merged**